src/openrct2/park/ParkFile.cpp

```diff
@@ -386,22 +399,107 @@ namespace OpenRCT2
 
                 ReadWriteStringTable(cs, gScenarioDetails, "en-GB");
 
-                cs.ReadWrite(gScenarioObjective.Type);
-                cs.ReadWrite(gScenarioObjective.Year);
-                cs.ReadWrite(gScenarioObjective.NumGuests);
-                cs.ReadWrite(gScenarioObjective.Currency);
-
-                cs.ReadWrite(gScenarioParkRatingWarningDays);
-
-                cs.ReadWrite(gScenarioCompletedCompanyValue);
-                if (gScenarioCompletedCompanyValue == MONEY64_UNDEFINED
-                    || gScenarioCompletedCompanyValue == COMPANY_VALUE_ON_FAILED_OBJECTIVE)
+                if (os.GetMode() == OrcaStream::Mode::READING)
                 {
-                    cs.Write("");
+                    cs.ReadWrite(gScenarioObjective.Type);
+                    uint8_t year;
+                    union
+                    {
+                        uint16_t numGuests;
+                        rct_string_id rideId;
```

Comment:
this is the wrong type for a rideId

Reply:
I literally copied that struct from the old scenario.h

Reply:
> I literally copied that struct from the old scenario.h

Unfortunately there are mistakes in the code base, we don't blame you for copying it. We are just suggesting you don't copy the mistake 🙂